x-pack/test_serverless/functional/services/ml/observability_navigation.ts

```diff
@@ -15,11 +15,11 @@ export function MachineLearningNavigationProviderObservability({
   const svlCommonNavigation = getPageObject('svlCommonNavigation');
 
   async function navigateToArea(id: string) {
-    await svlCommonNavigation.sidenav.openSection('observability_project_nav.aiops');
-    await testSubjects.existOrFail(`~nav-item-observability_project_nav.aiops.ml:${id}`, {
+    await svlCommonNavigation.sidenav.openPanel('machine_learning-landing');
+    await testSubjects.existOrFail(`~panelNavItem-id-ml:${id}`, {
       timeout: 60 * 1000,
     });
-    await testSubjects.click(`~nav-item-observability_project_nav.aiops.ml:${id}`);
+    await testSubjects.click(`~panelNavItem-id-ml:${id}`);
```

Comment:
Perhaps similar to line #19, is this a good place for a `missingOrFail()` call?

Reply:
@wayneseymour I see your point and makes sense what you say, I totally agree! FYI I didn't write these tests, I am adapting them to meet some new navigation changes (we replaced accordion with slideOpener).

I checked the [file](https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/functional/test_suites/observability/ml/anomaly_detection_jobs_list.ts) that is using the `navigateToAnomalyDetection` helper, and there are indeed quite a few [assertion calls](https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/functional/test_suites/observability/ml/anomaly_detection_jobs_list.ts#L47), so I think we should be fine.

I also checked `security` navigation tests and they use the same pattern. Navigation helpers find the nav items and click on them. Then the actual test files do all the assertions.

Reply:
Ok cool beans.  Have you tested this against MKI?

Reply:
No. Do you have any instructions on how I can test on MKI?

Reply:
Indeed!

https://docs.google.com/document/d/1tiax7xoDYwFXYZjRTgVKkVMjN-SQzBWk4yn1JY6Z5UY/edit?pli=1&tab=t.0#heading=h.f6esfmqe4nh5

Reply:
@wayneseymour I need some help here, I am stuck and I don't want to block this PR more. I tried to test against MKI using the approach my team shared with me, but with that approach it looks like tests are not being run from my local Kibana. Here's what I used:

```
TEST_CLOUD_HOST_NAME=console.qa.cld.elstc.co TEST_CLOUD=1 NODE_TLS_REJECT_UNAUTHORIZED=0 TEST_ES_URL="https://testing-internal/:NEW-PASSWORD@es-host:443" TEST_KIBANA_URL="https://testing-internal/:NEW-PASSWORD@kbn-host" node  --no-warnings scripts/functional_test_runner --exclude-tag skipMKI --exclude-tag failsOnMKI --config x-pack/test_serverless/functional/test_suites/observability/config.ts
```

For `es-host` and `kbn-host` I used the ones from the project I created on console.qa cloud and for `PASSWORD` I used the one I got calling the reset credential api. So when I used above command from my local branch to run navigation functional tests, I could easily tell that it is not reading my local kibana source code, since I could see the old navigation. 

I am trying now to follow the steps in the document you shared with me. I cloned the `qaf` repo locally. What do I do next? How do I use the qaf command?

Reply:
@wayneseymour I managed to set up `qaf` locally and I was able to run the tests against MKI using the docker image from my PR. Can I have the final approval for this PR?

Reply:
I ran serverless navigation and ml tests, and they are green!
<img width="500" alt="Screenshot 2024-11-08 at 14 02 09" src="https://github.com/user-attachments/assets/04d1aecb-21b2-4835-9702-cb210f838eb8">